src/vaadin-combo-box-data-provider-mixin.html

```diff
@@ -128,6 +128,21 @@
         return true;
       }
 
+      // Max item count in active range (items cache).
+      // This local var reflects the 'MAX_RANGE_COUNT' in combo box
+      // connector.
+      const MAX_RANGE_COUNT = Math.max(this.pageSize * 2, 500);
+
+      // We skip the page loading when the scrolling has been reset to 0
+      // position after assigning a new items to iron-list, because otherwise
+      // the connector's 'clearPageCallbacks' would be broken and the gaps in
+      // items list occur. Checking items cache size here, since the gaps only
+      // appear when the cache is fully populated and it should be possible for
+      // sure to load the page at the very beginning
+      if (this.$.overlay.resetScrolling && this.size > MAX_RANGE_COUNT) {
```

Comment:
Is the MAX_RANGE_COUNT check needed for anything? I'd imagine we could always skip loading pages when the scrolling is being reset?

Also, this if-clause should have test cases (for both conditions).

Additional though: Would it be possible to have this logic inside the wrapper module instead so that as much as possible of the scroll reset logic would be in the same place? If the conditions apply, the wrapper could skip dispatching the "index-requested" event.

Reply:
You're right, I've removed MAX_RANGE_COUNT and it seems to work still. I assume my previous issue which I tried to resolve by this limitation has been resolved by another code change.

This condition has been moved to wrapper, as you advised. Better to have it there.